src/pykeen/utils.py

```diff
@@ -509,6 +510,24 @@ def broadcast_cat(
     return torch.cat([x.repeat(*x_rep), y.repeat(*y_rep)], dim=dim)
 
 
+def multi_broadcast_cat(tensors: Iterable[torch.FloatTensor], dim: int) -> torch.FloatTensor:
+    """Apply :func:`broadcast_cat` to an arbitrary number of tensors.
+
+    :param tensors: A list of tensors to concatenate from left to right, using the same
+        semantics as :func:`torch.cat`
+    :param dim:
+        The concat dimension.
+
+    :return: A concatenated, broadcasted tensor.
+    """
+    tensors = iter(tensors)
+    first, second = next(tensors), next(tensors)
```

Comment:
When slightly restricting the generality from `Iterable` to `Sequence`, we could express these as `first, second = tensors[:2]`, which might be easier to read.

Reply:
Reasonable. There aren't a lot of good reasons to write code that could potentially handle big streams (now that I say it though, i'm tempted to do it just for the elegance)